apps/settings/js/modules/storage/volume.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * Volume:
+ *   - Volume is an Observable that wraps the platform DeviceStorage objects.
+ *   - It has some observable properties: name, isExternal, externalIndex,
+ *     isUnrecognised, availableState, volumeState, musicUsedSpace,
+ *     picturesUsedSpace, videosUsedSpace, sdcardUsedSpace, volumeFreeSpace.
+ * Volume only updates storage information and does not involve in any UI logic.
+ *
+ * @module Volume
+ */
+define(function(require) {
+  'use strict';
+
+  var Observable = require('modules/mvvm/observable');
+
+  const MEDIA_TYPE = ['music', 'pictures', 'videos', 'sdcard'];
+
+  var _debug = false;
+  var Debug = function() {};
+  if (_debug) {
+    Debug = function ms_debug(msg) {
+      console.log('--> [Volume]: ' + msg);
+    };
+  }
+
+  /**
+   * Provide a function to update latest storages used space.
+   * While the 'availableState' property is changed to be 'available',
+   * volume module will get the used space for all storages immediately.
+   *
+   * @access public
+   * @memberOf Volume
+   * @param {Object} evt
+   */
+  var updateStoragesSize = function() {
+    Debug('updateStoragesSize(): this.availableState = ' + this.availableState);
+    if (this.availableState !== 'available') {
+      return; // Early return if the 'availableState' is not 'available'.
+    }
+
+    MEDIA_TYPE.forEach((type) => {
+      // used space for each media type
+      this.data[type].usedSpace().then((size) => {
+        Debug('updateStoragesSize(): usedSpace(): size = ' +
+              JSON.stringify(size));
+        this[type + 'UsedSpace'] = size;
+      }, (reason) => {
+        Debug('updateStoragesSize(): usedSpace(): reason = ' + reason);
+      });
+    });
+    // free space remaining for this volume
+    this.data.sdcard.freeSpace().then((size) => {
+      Debug('updateStoragesSize(): freeSpace(): size = ' +
+            JSON.stringify(size));
+      this.volumeFreeSpace = size;
+    }, (reason) => {
+      Debug('updateStoragesSize(): freeSpace(): reason = ' + reason);
+    });
+  };
+
+  /**
+   * @class Volume
+   * @requires module:modules/mvvm/observable
+   * @param {Array} storages
```

Comment:
`storages` is not an array. It is an object looks like:
```
{
  "music":{},
  "pictures":{},
  "videos":{},
  "sdcard":{}
}
```
We can add this to the comment as there are confusing namings in the gecko API.